src/sql/workbench/contrib/notebook/browser/models/notebookModel.ts

```diff
@@ -726,14 +708,9 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	private refreshConnections(newConnection: ConnectionProfile) {
 		if (this.isValidConnection(newConnection) &&
 			this._activeConnection.id !== '-1' &&
-			this._activeConnection.id !== this._activeContexts.defaultConnection.id) {
-			// Put the defaultConnection to the head of otherConnections
-			if (this.isValidConnection(this._activeContexts.defaultConnection)) {
-				this._activeContexts.otherConnections = this._activeContexts.otherConnections.filter(conn => conn.id !== this._activeContexts.defaultConnection.id);
-				this._activeContexts.otherConnections.unshift(this._activeContexts.defaultConnection);
-			}
+			this._activeConnection.id !== newConnection.id) {
 			// Change the defaultConnection to newConnection
-			this._activeContexts.defaultConnection = newConnection;
+			this._activeConnection = newConnection;
 		}
```

Comment:
The comment here isn't correct anymore. This whole method is weird anyways, I'm not sure what it's doing since the name (refreshConnections) doesn't seem to match the logic.